great_expectations/core/batch.py

```diff
@@ -718,21 +721,24 @@ def get_batch_request_from_acceptable_arguments(
             # Raise a warning if kwargs exist
             pass
 
-        batch_request = RuntimeBatchRequest(
-            datasource_name=datasource_name,
-            data_connector_name=data_connector_name,
-            data_asset_name=data_asset_name,
-            runtime_parameters=runtime_parameters,
-            batch_identifiers=batch_identifiers,
-            batch_spec_passthrough=batch_spec_passthrough,
-        )
+        batch_request_as_dict = {
+            "datasource_name": datasource_name,
+            "data_connector_name": data_connector_name,
+            "data_asset_name": data_asset_name,
+            "runtime_parameters": runtime_parameters,
+            "batch_identifiers": batch_identifiers,
+            "batch_spec_passthrough": batch_spec_passthrough,
+        }
     else:
+        batch_request_class = BatchRequest
+
         if data_connector_query is None:
             if batch_filter_parameters is not None and batch_identifiers is not None:
                 raise ValueError(
                     'Must provide either "batch_filter_parameters" or "batch_identifiers", not both.'
                 )
-            elif batch_filter_parameters is None and batch_identifiers is not None:
+
+            if batch_filter_parameters is None and batch_identifiers is not None:
```

Comment:
How can ifs on lines 736 and 741 both be True? It would seem like `elif` is the appropriate flow control.

Reply:
@NathanFarmer If line 736 evaluates to `True`, exception will be raised and like 741 will not be reached.  Thanks.